yaml/stage.go

```diff
@@ -79,7 +79,7 @@ func (s *StageSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 
 		// implicitly set the stage `needs` if empty
-		if len(stage.Needs) == 0 {
+		if len(stage.Needs) == 0 && stage.Name != "clone" && stage.Name != "init" {
```

Comment:
Ensures that doing `yaml.Marshal` on the `Stage` struct does not incorrectly add the `clone` needs to the `clone` and `init` stage.